sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/AutoConfiguredOpenTelemetrySdkBuilder.java

```diff
@@ -403,41 +409,53 @@ public AutoConfiguredOpenTelemetrySdk build() {
       boolean sdkEnabled = !config.getBoolean("otel.sdk.disabled", false);
 
       if (sdkEnabled) {
-        SdkMeterProviderBuilder meterProviderBuilder = SdkMeterProvider.builder();
-        meterProviderBuilder.setResource(resource);
-        MeterProviderConfiguration.configureMeterProvider(
-            meterProviderBuilder, config, spiHelper, metricExporterCustomizer, closeables);
-        meterProviderBuilder = meterProviderCustomizer.apply(meterProviderBuilder, config);
-        SdkMeterProvider meterProvider = meterProviderBuilder.build();
+        SdkMeterProvider meterProvider =
+            spiHelper.loadOptional(SdkMeterProvider.class).orElse(null);
```

Comment:
Need to think about this. There is no SPI for providing a custom SdkMeterProvider today - only the ability to fully customize / replace the SdkMeterProviderBuilder via `AutoConfigurationCustomizer#addMeterProviderCustomizer`. So the action of `spiHelper.loadOptional(SdkMeterProvider.class)` will never have any effect outside the spring boot autoconfigure use case.

(Same applies to the customizations below.)

Reply:
Very good point!

Spring knows both styles - replacing beans and customizing beans.
The current spring starter only supports replacing, but we're making breaking changes anyways, so that's not a strong argument. 

Below are examples for both styles - as a spring user would write it. @jeanbisutti do you have a preference?

**Replacing a bean:**

```java
@Bean 
public SdkTracerProvider tracerProvider() {
  return SdkTracerProviderBuilder tracerProviderBuilder = SdkTracerProvider.builder()...build(); 
}
```

**Customizing a bean:**

Note: For technical reasons, the bi-functions probably has to be put in an interface.

```java
@Bean 
public BiFunction<SdkLoggerProviderBuilder, ConfigProperties, SdkLoggerProviderBuilder> tracerProviderCustomizer() { 
  return (a, config) -> a..;
}
```

Reply:
Be able to replace Spring bean is something I would like to keep. It seems something usual in the Spring Boot code. It also help provide workarounds in case of bugs. In addition, it can help companies to provide a custom distribution of the OTel Spring starter.

Be able to provide `SpanExporterCustomizer`, `MetricExporterCustomizer` and `LogRecordExporterCustomizer` would improve the extensibility for users and companies providing a distribution of the OTel Spring starter.

Reply:
@jack-berg I've discussed with @jeanbisutti - turns out the customizer is sufficient for the use cases we can think of.